TestWorker/delete-rule-with-async-delete-interval is flaky when run in Github CI

**Describe the bug**

I have observed the test fail a couple of time on master. The full log is below:

```text
2020-11-12T21:04:21.6786231Z --- FAIL: TestWorker (0.02s)
2020-11-12T21:04:21.6787576Z     --- FAIL: TestWorker/delete-rule-with-async-delete-interval (0.01s)
2020-11-12T21:04:21.6795522Z ##[error]        allocator_test.go:226: Expect the rule with id 1 to be deleted from async rule cache
2020-11-12T21:04:21.6806938Z I1112 21:03:59.645378   11984 networkpolicy_controller.go:363] Waiting for Antrea client to be ready
2020-11-12T21:04:21.6807751Z I1112 21:03:59.645582   11984 networkpolicy_controller.go:374] Antrea client is ready
2020-11-12T21:04:21.6808481Z I1112 21:03:59.645975   11984 networkpolicy_controller.go:383] Waiting for all watchers to complete full sync
2020-11-12T21:04:21.6809582Z I1112 21:03:59.646066   11984 networkpolicy_controller.go:564] Starting watch for NetworkPolicy
2020-11-12T21:04:21.6810517Z I1112 21:03:59.646183   11984 networkpolicy_controller.go:571] Started watch for NetworkPolicy
2020-11-12T21:04:21.6811345Z I1112 21:03:59.646229   11984 networkpolicy_controller.go:601] Received 1 init events for NetworkPolicy
2020-11-12T21:04:21.6812382Z I1112 21:03:59.646298   11984 networkpolicy_controller.go:191] NetworkPolicy K8sNetworkPolicy:ns1/policy1 applied to Pods on this Node
2020-11-12T21:04:21.6813471Z I1112 21:03:59.646594   11984 networkpolicy_controller.go:564] Starting watch for AddressGroup
2020-11-12T21:04:21.6814166Z I1112 21:03:59.646697   11984 networkpolicy_controller.go:571] Started watch for AddressGroup
2020-11-12T21:04:21.6814874Z I1112 21:03:59.646823   11984 networkpolicy_controller.go:564] Starting watch for AppliedToGroup
2020-11-12T21:04:21.6815597Z I1112 21:03:59.646958   11984 networkpolicy_controller.go:571] Started watch for AppliedToGroup
2020-11-12T21:04:21.6816505Z I1112 21:03:59.746614   11984 networkpolicy_controller.go:601] Received 1 init events for AddressGroup
2020-11-12T21:04:21.6817323Z I1112 21:03:59.846958   11984 networkpolicy_controller.go:601] Received 1 init events for AppliedToGroup
2020-11-12T21:04:21.6818162Z I1112 21:03:59.847104   11984 networkpolicy_controller.go:385] All watchers have completed full sync, installing flows for init events
2020-11-12T21:04:21.6818993Z I1112 21:03:59.847199   11984 networkpolicy_controller.go:389] Starting NetworkPolicy workers now
2020-11-12T21:04:21.6819808Z I1112 21:03:59.847576   11984 networkpolicy_controller.go:395] Starting IDAllocator worker to maintain the async rule cache
2020-11-12T21:04:21.6820608Z I1112 21:03:59.847786   11984 status_controller.go:209] Starting NetworkPolicy StatusController
2020-11-12T21:04:21.6821527Z I1112 21:03:59.848387   11984 status_controller.go:216] Shutting down NetworkPolicy StatusController
2020-11-12T21:04:21.6822253Z I1112 21:03:59.850434   11984 networkpolicy_controller.go:363] Waiting for Antrea client to be ready
2020-11-12T21:04:21.6822907Z I1112 21:03:59.850492   11984 networkpolicy_controller.go:374] Antrea client is ready
2020-11-12T21:04:21.6823585Z I1112 21:03:59.850621   11984 networkpolicy_controller.go:564] Starting watch for AppliedToGroup
2020-11-12T21:04:21.6824307Z I1112 21:03:59.850704   11984 networkpolicy_controller.go:571] Started watch for AppliedToGroup
2020-11-12T21:04:21.6825024Z I1112 21:03:59.850839   11984 networkpolicy_controller.go:564] Starting watch for AddressGroup
2020-11-12T21:04:21.6825909Z I1112 21:03:59.850928   11984 networkpolicy_controller.go:571] Started watch for AddressGroup
2020-11-12T21:04:21.6826622Z I1112 21:03:59.851001   11984 networkpolicy_controller.go:601] Received 1 init events for AddressGroup
2020-11-12T21:04:21.6827353Z I1112 21:03:59.851081   11984 networkpolicy_controller.go:601] Received 1 init events for AppliedToGroup
2020-11-12T21:04:21.6828074Z I1112 21:03:59.851808   11984 networkpolicy_controller.go:564] Starting watch for NetworkPolicy
2020-11-12T21:04:21.6828770Z I1112 21:03:59.851907   11984 networkpolicy_controller.go:571] Started watch for NetworkPolicy
2020-11-12T21:04:21.6829490Z I1112 21:03:59.851957   11984 networkpolicy_controller.go:601] Received 1 init events for NetworkPolicy
2020-11-12T21:04:21.6830337Z I1112 21:03:59.852019   11984 networkpolicy_controller.go:191] NetworkPolicy K8sNetworkPolicy:ns1/policy1 applied to Pods on this Node
2020-11-12T21:04:21.6831190Z I1112 21:03:59.852247   11984 networkpolicy_controller.go:383] Waiting for all watchers to complete full sync
2020-11-12T21:04:21.6832376Z I1112 21:03:59.852283   11984 networkpolicy_controller.go:385] All watchers have completed full sync, installing flows for init events
2020-11-12T21:04:21.6838498Z E1112 21:03:59.852405   11984 networkpolicy_controller.go:498] Rule 8af48c355979fb68 is not complete or does not exist in cache
2020-11-12T21:04:21.6839442Z I1112 21:03:59.852452   11984 networkpolicy_controller.go:389] Starting NetworkPolicy workers now
2020-11-12T21:04:21.6840997Z I1112 21:03:59.853003   11984 networkpolicy_controller.go:395] Starting IDAllocator worker to maintain the async rule cache
2020-11-12T21:04:21.6841764Z I1112 21:03:59.853288   11984 status_controller.go:209] Starting NetworkPolicy StatusController
2020-11-12T21:04:21.6842479Z I1112 21:04:00.052698   11984 status_controller.go:216] Shutting down NetworkPolicy StatusController
2020-11-12T21:04:21.6843342Z I1112 21:04:00.053440   11984 networkpolicy_controller.go:363] Waiting for Antrea client to be ready
2020-11-12T21:04:21.6844492Z I1112 21:04:00.053489   11984 networkpolicy_controller.go:374] Antrea client is ready
2020-11-12T21:04:21.6845334Z I1112 21:04:00.053686   11984 networkpolicy_controller.go:564] Starting watch for AppliedToGroup
2020-11-12T21:04:21.6847249Z I1112 21:04:00.053779   11984 networkpolicy_controller.go:383] Waiting for all watchers to complete full sync
2020-11-12T21:04:21.6848331Z I1112 21:04:00.053785   11984 networkpolicy_controller.go:571] Started watch for AppliedToGroup
2020-11-12T21:04:21.6849358Z I1112 21:04:00.053838   11984 networkpolicy_controller.go:564] Starting watch for NetworkPolicy
2020-11-12T21:04:21.6850654Z I1112 21:04:00.053922   11984 networkpolicy_controller.go:571] Started watch for NetworkPolicy
2020-11-12T21:04:21.6851239Z I1112 21:04:00.053847   11984 networkpolicy_controller.go:564] Starting watch for AddressGroup
2020-11-12T21:04:21.6851816Z I1112 21:04:00.054004   11984 networkpolicy_controller.go:571] Started watch for AddressGroup
2020-11-12T21:04:21.6852431Z I1112 21:04:00.054080   11984 networkpolicy_controller.go:601] Received 1 init events for AppliedToGroup
2020-11-12T21:04:21.6853342Z I1112 21:04:00.054088   11984 networkpolicy_controller.go:601] Received 1 init events for AddressGroup
2020-11-12T21:04:21.6853969Z I1112 21:04:00.054171   11984 networkpolicy_controller.go:601] Received 1 init events for NetworkPolicy
2020-11-12T21:04:21.6854695Z I1112 21:04:00.054221   11984 networkpolicy_controller.go:191] NetworkPolicy K8sNetworkPolicy:ns1/policy1 applied to Pods on this Node
2020-11-12T21:04:21.6855490Z I1112 21:04:00.054463   11984 networkpolicy_controller.go:385] All watchers have completed full sync, installing flows for init events
2020-11-12T21:04:21.6856175Z I1112 21:04:00.054567   11984 networkpolicy_controller.go:389] Starting NetworkPolicy workers now
2020-11-12T21:04:21.6856919Z I1112 21:04:00.054692   11984 networkpolicy_controller.go:175] NetworkPolicy K8sNetworkPolicy:ns1/policy1 no longer applied to Pods on this Node
2020-11-12T21:04:21.6857859Z I1112 21:04:00.055143   11984 networkpolicy_controller.go:395] Starting IDAllocator worker to maintain the async rule cache
2020-11-12T21:04:21.6858538Z I1112 21:04:00.055483   11984 status_controller.go:209] Starting NetworkPolicy StatusController
2020-11-12T21:04:21.6859182Z I1112 21:04:00.056016   11984 status_controller.go:216] Shutting down NetworkPolicy StatusController
2020-11-12T21:04:21.6859796Z I1112 21:04:00.056266   11984 networkpolicy_controller.go:363] Waiting for Antrea client to be ready
2020-11-12T21:04:21.6860513Z I1112 21:04:00.056329   11984 networkpolicy_controller.go:374] Antrea client is ready
2020-11-12T21:04:21.6861065Z I1112 21:04:00.056425   11984 networkpolicy_controller.go:564] Starting watch for AppliedToGroup
2020-11-12T21:04:21.6861665Z I1112 21:04:00.056471   11984 networkpolicy_controller.go:383] Waiting for all watchers to complete full sync
2020-11-12T21:04:21.6862445Z I1112 21:04:00.056502   11984 networkpolicy_controller.go:571] Started watch for AppliedToGroup
2020-11-12T21:04:21.6863227Z I1112 21:04:00.056532   11984 networkpolicy_controller.go:564] Starting watch for AddressGroup
2020-11-12T21:04:21.6864077Z I1112 21:04:00.056576   11984 networkpolicy_controller.go:564] Starting watch for NetworkPolicy
2020-11-12T21:04:21.6864697Z I1112 21:04:00.056621   11984 networkpolicy_controller.go:571] Started watch for AddressGroup
2020-11-12T21:04:21.6865313Z I1112 21:04:00.056660   11984 networkpolicy_controller.go:571] Started watch for NetworkPolicy
2020-11-12T21:04:21.6866346Z I1112 21:04:00.056829   11984 networkpolicy_controller.go:601] Received 2 init events for AddressGroup
2020-11-12T21:04:21.6867184Z I1112 21:04:00.056886   11984 networkpolicy_controller.go:601] Received 1 init events for NetworkPolicy
2020-11-12T21:04:21.6867932Z I1112 21:04:00.056983   11984 networkpolicy_controller.go:191] NetworkPolicy K8sNetworkPolicy:ns1/policy1 applied to Pods on this Node
2020-11-12T21:04:21.6868875Z I1112 21:04:00.056918   11984 networkpolicy_controller.go:601] Received 1 init events for AppliedToGroup
2020-11-12T21:04:21.6869769Z I1112 21:04:00.057524   11984 networkpolicy_controller.go:385] All watchers have completed full sync, installing flows for init events
2020-11-12T21:04:21.6870471Z I1112 21:04:00.057718   11984 networkpolicy_controller.go:389] Starting NetworkPolicy workers now
2020-11-12T21:04:21.6871161Z I1112 21:04:00.057992   11984 networkpolicy_controller.go:395] Starting IDAllocator worker to maintain the async rule cache
2020-11-12T21:04:21.6872031Z I1112 21:04:00.058174   11984 status_controller.go:209] Starting NetworkPolicy StatusController
2020-11-12T21:04:21.6873096Z I1112 21:04:00.060378   11984 status_controller.go:216] Shutting down NetworkPolicy StatusController
2020-11-12T21:04:21.6873999Z I1112 21:04:00.061338   11984 networkpolicy_controller.go:363] Waiting for Antrea client to be ready
2020-11-12T21:04:21.6874591Z I1112 21:04:00.061408   11984 networkpolicy_controller.go:374] Antrea client is ready
2020-11-12T21:04:21.6875532Z I1112 21:04:00.061581   11984 networkpolicy_controller.go:383] Waiting for all watchers to complete full sync
2020-11-12T21:04:21.6876317Z I1112 21:04:00.061600   11984 networkpolicy_controller.go:564] Starting watch for AddressGroup
2020-11-12T21:04:21.6877079Z I1112 21:04:00.061660   11984 networkpolicy_controller.go:564] Starting watch for AppliedToGroup
2020-11-12T21:04:21.6877653Z I1112 21:04:00.061785   11984 networkpolicy_controller.go:571] Started watch for AddressGroup
2020-11-12T21:04:21.6878222Z I1112 21:04:00.062238   11984 networkpolicy_controller.go:564] Starting watch for NetworkPolicy
2020-11-12T21:04:21.6878815Z I1112 21:04:00.062259   11984 networkpolicy_controller.go:601] Received 1 init events for AddressGroup
2020-11-12T21:04:21.6879403Z I1112 21:04:00.062336   11984 networkpolicy_controller.go:571] Started watch for NetworkPolicy
2020-11-12T21:04:21.6879980Z I1112 21:04:00.061796   11984 networkpolicy_controller.go:571] Started watch for AppliedToGroup
2020-11-12T21:04:21.6880841Z I1112 21:04:00.062517   11984 networkpolicy_controller.go:601] Received 1 init events for AppliedToGroup
2020-11-12T21:04:21.6881473Z I1112 21:04:00.062591   11984 networkpolicy_controller.go:601] Received 1 init events for NetworkPolicy
2020-11-12T21:04:21.6882198Z I1112 21:04:00.062700   11984 networkpolicy_controller.go:191] NetworkPolicy K8sNetworkPolicy:ns1/policy1 applied to Pods on this Node
2020-11-12T21:04:21.6882983Z I1112 21:04:00.062982   11984 networkpolicy_controller.go:385] All watchers have completed full sync, installing flows for init events
2020-11-12T21:04:21.6883850Z I1112 21:04:00.063151   11984 networkpolicy_controller.go:389] Starting NetworkPolicy workers now
2020-11-12T21:04:21.6884482Z I1112 21:04:00.064172   11984 networkpolicy_controller.go:395] Starting IDAllocator worker to maintain the async rule cache
2020-11-12T21:04:21.6885114Z I1112 21:04:00.065757   11984 status_controller.go:209] Starting NetworkPolicy StatusController
2020-11-12T21:04:21.6885680Z E1112 21:04:00.084311   11984 networkpolicy_controller.go:640] Unknown event: {BOOKMARK <nil>}
2020-11-12T21:04:21.6886275Z I1112 21:04:00.084420   11984 networkpolicy_controller.go:575] Stopped watch for AddressGroup, total items received: 2
2020-11-12T21:04:21.6886866Z E1112 21:04:00.084540   11984 networkpolicy_controller.go:640] Unknown event: {BOOKMARK <nil>}
2020-11-12T21:04:21.6887473Z I1112 21:04:00.084608   11984 networkpolicy_controller.go:575] Stopped watch for AppliedToGroup, total items received: 2
2020-11-12T21:04:21.6888262Z I1112 21:04:00.084940   11984 networkpolicy_controller.go:148] NetworkPolicy K8sNetworkPolicy:ns1/policy2 applied to Pods on this Node
2020-11-12T21:04:21.6889079Z I1112 21:04:00.111977   11984 networkpolicy_controller.go:175] NetworkPolicy K8sNetworkPolicy:ns1/policy1 no longer applied to Pods on this Node
2020-11-12T21:04:21.6889890Z I1112 21:04:00.123983   11984 networkpolicy_controller.go:175] NetworkPolicy K8sNetworkPolicy:ns1/policy2 no longer applied to Pods on this Node
2020-11-12T21:04:21.6890607Z I1112 21:04:00.135255   11984 status_controller.go:216] Shutting down NetworkPolicy StatusController
2020-11-12T21:04:21.6891545Z I1112 21:04:17.789871   11984 reconciler.go:710] Forgetting rule unknown-rule-id
2020-11-12T21:04:21.6892000Z I1112 21:04:17.790264   11984 reconciler.go:710] Forgetting rule foo
2020-11-12T21:04:21.6892378Z I1112 21:04:17.790702   11984 reconciler.go:710] Forgetting rule foo
2020-11-12T21:04:21.6896273Z I1112 21:04:17.791189   11984 reconciler.go:710] Forgetting rule foo
2020-11-12T21:04:21.6897195Z I1112 21:04:17.792508   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6898360Z I1112 21:04:17.793593   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6899186Z I1112 21:04:17.794521   11984 reconciler.go:750] Can't find interface for Pod ns1/pod2, skipping
2020-11-12T21:04:21.6900027Z I1112 21:04:17.795210   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6901382Z I1112 21:04:17.796513   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6902778Z I1112 21:04:17.798105   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6903771Z W1112 21:04:17.799067   11984 reconciler.go:925] Can not resolve port http for endpoints &GroupMember{Pod:&PodReference{Name:pod1,Namespace:ns1,},ExternalEntity:nil,IPs:[],Ports:[]NamedPort{},}
2020-11-12T21:04:21.6905098Z I1112 21:04:17.799911   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6906420Z I1112 21:04:17.801363   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6907344Z I1112 21:04:17.802083   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6908450Z I1112 21:04:17.802758   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6909359Z I1112 21:04:17.803334   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6910453Z I1112 21:04:17.804062   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6911362Z E1112 21:04:17.805891   11984 reconciler.go:336] rule ingress-rule already realized during the initialization phase
2020-11-12T21:04:21.6912362Z I1112 21:04:17.807462   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6913588Z I1112 21:04:17.813421   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6914551Z I1112 21:04:17.814240   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6915475Z I1112 21:04:17.814501   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6916583Z I1112 21:04:17.815311   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6917934Z I1112 21:04:17.815569   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6919074Z I1112 21:04:17.815661   11984 reconciler.go:750] Can't find interface for Pod ns1/pod4, skipping
2020-11-12T21:04:21.6920072Z I1112 21:04:17.816371   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6921847Z I1112 21:04:17.816593   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6923017Z I1112 21:04:17.816696   11984 reconciler.go:770] Can't find interface for Pod ns1/pod4, skipping
2020-11-12T21:04:21.6924240Z I1112 21:04:17.817488   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6925413Z I1112 21:04:17.817709   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6926871Z I1112 21:04:17.818405   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy AntreaClusterNetworkPolicy:name1
2020-11-12T21:04:21.6928593Z I1112 21:04:17.818755   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy AntreaClusterNetworkPolicy:name1
2020-11-12T21:04:21.6929960Z I1112 21:04:17.819394   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy AntreaClusterNetworkPolicy:name1
2020-11-12T21:04:21.6931110Z I1112 21:04:17.819807   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy AntreaClusterNetworkPolicy:name1
2020-11-12T21:04:21.6932305Z W1112 21:04:17.823885   11984 reconciler.go:925] Can not resolve port https for endpoints &GroupMember{Pod:nil,ExternalEntity:nil,IPs:[[0 0 0 0 0 0 0 0 0 0 255 255 1 1 1 2]],Ports:[]NamedPort{NamedPort{Port:80,Name:http,Protocol:TCP,},},}
2020-11-12T21:04:21.6957979Z W1112 21:04:17.824043   11984 reconciler.go:925] Can not resolve port https for endpoints &GroupMember{Pod:nil,ExternalEntity:nil,IPs:[[0 0 0 0 0 0 0 0 0 0 255 255 1 1 1 3]],Ports:[]NamedPort{NamedPort{Port:8080,Name:http,Protocol:TCP,},},}
2020-11-12T21:04:21.6959293Z W1112 21:04:17.824249   11984 reconciler.go:925] Can not resolve port http for endpoints &GroupMember{Pod:nil,ExternalEntity:nil,IPs:[[0 0 0 0 0 0 0 0 0 0 255 255 1 1 1 4]],Ports:[]NamedPort{NamedPort{Port:443,Name:https,Protocol:TCP,},},}
2020-11-12T21:04:21.6960264Z W1112 21:04:17.824358   11984 reconciler.go:925] Can not resolve port http for endpoints &GroupMember{Pod:nil,ExternalEntity:nil,IPs:[[0 0 0 0 0 0 0 0 0 0 255 255 1 1 1 5]],Ports:[]NamedPort{},}
2020-11-12T21:04:21.6961281Z W1112 21:04:17.824429   11984 reconciler.go:925] Can not resolve port https for endpoints &GroupMember{Pod:nil,ExternalEntity:nil,IPs:[[0 0 0 0 0 0 0 0 0 0 255 255 1 1 1 5]],Ports:[]NamedPort{},}
2020-11-12T21:04:21.6962240Z W1112 21:04:17.824506   11984 reconciler.go:925] Can not resolve port http for endpoints &GroupMember{Pod:nil,ExternalEntity:nil,IPs:[[0 0 0 0 0 0 0 0 0 0 255 255 1 1 1 6]],Ports:[]NamedPort{NamedPort{Port:443,Name:foo,Protocol:TCP,},},}
2020-11-12T21:04:21.6963291Z W1112 21:04:17.824596   11984 reconciler.go:925] Can not resolve port https for endpoints &GroupMember{Pod:nil,ExternalEntity:nil,IPs:[[0 0 0 0 0 0 0 0 0 0 255 255 1 1 1 6]],Ports:[]NamedPort{NamedPort{Port:443,Name:foo,Protocol:TCP,},},}
2020-11-12T21:04:21.6964338Z W1112 21:04:17.824774   11984 reconciler.go:925] Can not resolve port https for endpoints &GroupMember{Pod:nil,ExternalEntity:nil,IPs:[[0 0 0 0 0 0 0 0 0 0 255 255 1 1 1 1]],Ports:[]NamedPort{NamedPort{Port:80,Name:http,Protocol:TCP,},},}
2020-11-12T21:04:21.6965706Z I1112 21:04:17.826743   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6966900Z I1112 21:04:17.827688   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6967730Z I1112 21:04:17.827827   11984 reconciler.go:750] Can't find interface for Pod ns1/pod2, skipping
2020-11-12T21:04:21.6968566Z I1112 21:04:17.828490   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6969641Z I1112 21:04:17.829807   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6970839Z I1112 21:04:17.830860   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6971724Z W1112 21:04:17.831048   11984 reconciler.go:925] Can not resolve port http for endpoints &GroupMember{Pod:&PodReference{Name:pod1,Namespace:ns1,},ExternalEntity:nil,IPs:[],Ports:[]NamedPort{},}
2020-11-12T21:04:21.6972918Z I1112 21:04:17.832076   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6973875Z I1112 21:04:17.833059   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6974789Z I1112 21:04:17.833937   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6975706Z I1112 21:04:17.834597   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6976620Z I1112 21:04:17.835342   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6977736Z I1112 21:04:17.836472   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6978691Z I1112 21:04:17.837209   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6979383Z I1112 21:04:17.837712   11984 reconciler.go:851] IPBlock 10.10.0.0/16 is using unsupported address family, skip it
2020-11-12T21:04:21.6980418Z I1112 21:04:17.838828   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6983484Z I1112 21:04:17.840326   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6984487Z I1112 21:04:17.840787   11984 reconciler.go:750] Can't find interface for Pod ns1/pod2, skipping
2020-11-12T21:04:21.6985518Z I1112 21:04:17.841531   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6986848Z I1112 21:04:17.842970   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6987818Z I1112 21:04:17.843666   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6988901Z W1112 21:04:17.843853   11984 reconciler.go:925] Can not resolve port http for endpoints &GroupMember{Pod:&PodReference{Name:pod1,Namespace:ns1,},ExternalEntity:nil,IPs:[],Ports:[]NamedPort{},}
2020-11-12T21:04:21.6990584Z I1112 21:04:17.844524   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6992888Z I1112 21:04:17.845236   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6994341Z I1112 21:04:17.846099   11984 reconciler.go:234] Reconciling rule ingress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6996207Z I1112 21:04:17.846731   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6997676Z I1112 21:04:17.847499   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6998643Z I1112 21:04:17.848728   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.6999598Z I1112 21:04:17.849438   11984 reconciler.go:234] Reconciling rule egress-rule of NetworkPolicy K8sNetworkPolicy:ns1/name1
2020-11-12T21:04:21.7000307Z I1112 21:04:17.850520   11984 status_controller.go:209] Starting NetworkPolicy StatusController
2020-11-12T21:04:21.7001919Z I1112 21:04:18.350677   11984 status_controller.go:216] Shutting down NetworkPolicy StatusController
2020-11-12T21:04:21.7002762Z I1112 21:04:18.351717   11984 status_controller.go:209] Starting NetworkPolicy StatusController
2020-11-12T21:04:21.7003442Z I1112 21:04:18.851873   11984 status_controller.go:216] Shutting down NetworkPolicy StatusController
2020-11-12T21:04:21.7004271Z I1112 21:04:18.852700   11984 status_controller.go:209] Starting NetworkPolicy StatusController
2020-11-12T21:04:21.7004890Z I1112 21:04:19.353664   11984 status_controller.go:216] Shutting down NetworkPolicy StatusController
2020-11-12T21:04:21.7005502Z I1112 21:04:19.353756   11984 status_controller.go:209] Starting NetworkPolicy StatusController
2020-11-12T21:04:21.7006116Z I1112 21:04:21.657305   11984 status_controller.go:216] Shutting down NetworkPolicy StatusController
2020-11-12T21:04:21.7006513Z FAIL
```

--- a/pkg/agent/controller/networkpolicy/allocator_test.go
+++ b/pkg/agent/controller/networkpolicy/allocator_test.go
@@ -28,7 +28,8 @@ import (
 )
 
 var (
-	testDeleteInterval = 5 * time.Millisecond
+	testAsyncDeleteInterval    = 50 * time.Millisecond
+	testMinAsyncDeleteInterval = 100 * time.Millisecond
 )
 
 func TestNewIDAllocator(t *testing.T) {
@@ -63,7 +64,8 @@ func TestNewIDAllocator(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			got := newIDAllocator(testDeleteInterval, tt.args...)
+			minAsyncDeleteInterval = testMinAsyncDeleteInterval
+			got := newIDAllocator(testAsyncDeleteInterval, tt.args...)
 			assert.Equalf(t, tt.expectedLastAllocatedID, got.lastAllocatedID, "Got lastAllocatedID %v, expected %v", got.lastAllocatedID, tt.expectedLastAllocatedID)
 			assert.Equalf(t, tt.expectedAvailableSets, got.availableSet, "Got availableSet %v, expected %v", got.availableSet, tt.expectedAvailableSets)
 			assert.Equalf(t, tt.expectedAvailableSlice, got.availableSlice, "Got availableSlice %v, expected %v", got.availableSlice, tt.expectedAvailableSlice)
@@ -109,7 +111,8 @@ func TestAllocateForRule(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			a := newIDAllocator(testDeleteInterval, tt.args...)
+			minAsyncDeleteInterval = testMinAsyncDeleteInterval
+			a := newIDAllocator(testAsyncDeleteInterval, tt.args...)
 			actualErr := a.allocateForRule(tt.rule)
 			if actualErr != tt.expectedErr {
 				t.Fatalf("Got error %v, expected %v", actualErr, tt.expectedErr)
@@ -159,7 +162,8 @@ func TestRelease(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			a := newIDAllocator(testDeleteInterval, tt.newArgs...)
+			minAsyncDeleteInterval = testMinAsyncDeleteInterval
+			a := newIDAllocator(testAsyncDeleteInterval, tt.newArgs...)
 			actualErr := a.release(tt.releaseArgs)
 			assert.Equalf(t, tt.expectedErr, actualErr, "Got error %v, expected %v", actualErr, tt.expectedErr)
 			assert.Equalf(t, tt.expectedAvailableSets, a.availableSet, "Got availableSet %v, expected %v", a.availableSet, tt.expectedAvailableSets)
@@ -168,7 +172,7 @@ func TestRelease(t *testing.T) {
 	}
 }
 
-func TestWorker(t *testing.T) {
+func TestIdAllocatorWorker(t *testing.T) {
 	rule := &types.PolicyRule{
 		Direction: v1beta2.DirectionIn,
 		From:      []types.Address{},
@@ -176,25 +180,29 @@ func TestWorker(t *testing.T) {
 		Service:   nil,
 	}
 	tests := []struct {
-		name              string
-		args              []uint32
-		minDeleteInterval time.Duration
-		rule              *types.PolicyRule
-		expectedID        uint32
-		expectedErr       error
+		name                    string
+		args                    []uint32
+		testAsyncDeleteInterval time.Duration
+		rule                    *types.PolicyRule
+		expectedID              uint32
+		expectedErr             error
 	}{
 		{
-			"delete-rule-with-async-delete-interval",
+			// testMinAsyncDeleteInterval(100ms) is larger than testAsyncDeleteInterval(50ms),
+			// so rule should take at least 100ms to be deleted.
+			"delete-rule-with-test-min-async-delete-interval",
 			nil,
-			5 * time.Millisecond,
+			50 * time.Millisecond,
 			rule,
 			1,
 			nil,
 		},
 		{
-			"delete-rule-with-flow-poll-interval",
+			// testAsyncDeleteInterval(200ms) is larger than testMinAsyncDeleteInterval(100ms),
+			// so rule should take at least 200ms to be deleted.
+			"delete-rule-with-test-async-delete-interval",
 			nil,
-			1 * time.Millisecond,
+			200 * time.Millisecond,
 			rule,
 			1,
 			nil,
@@ -202,35 +210,43 @@ func TestWorker(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			minAsyncDeleteInterval = tt.minDeleteInterval
-			a := newIDAllocator(testDeleteInterval, tt.args...)
+			minAsyncDeleteInterval = testMinAsyncDeleteInterval
+			testAsyncDeleteInterval = tt.testAsyncDeleteInterval
+			a := newIDAllocator(testAsyncDeleteInterval, tt.args...)
 			actualErr := a.allocateForRule(tt.rule)
 			if actualErr != tt.expectedErr {
 				t.Fatalf("Got error %v, expected %v", actualErr, tt.expectedErr)
 			}
 			stopCh := make(chan struct{})
 			defer close(stopCh)
-			go wait.Until(a.worker, time.Millisecond, stopCh)
+			defer a.deleteQueue.ShutDown()
+			go wait.Until(a.worker, 50*time.Millisecond, stopCh)
 
-			start := time.Now()
 			a.forgetRule(tt.rule.FlowID)
+
+			startTime := time.Now()
+			var elapsedTime time.Duration
 			conditionFunc := func() (bool, error) {
 				a.Lock()
 				defer a.Unlock()
+				if startTime.IsZero() {
+					startTime = time.Now()
+				}
 				if len(a.availableSlice) > 0 {
+					elapsedTime = time.Since(startTime)
 					return true, nil
 				}
 				return false, nil
 			}
-			if err := wait.PollImmediate(time.Millisecond, time.Millisecond*10, conditionFunc); err != nil {
+
+			if err := wait.PollImmediate(50*time.Millisecond, time.Second, conditionFunc); err != nil {
 				t.Fatalf("Expect the rule with id %v to be deleted from async rule cache", tt.expectedID)
 			}
 			_, exists, err := a.getRuleFromAsyncCache(tt.expectedID)
-			assert.Falsef(t, exists, "Expect rule to be not present in asyncRuleCache")
+			assert.Falsef(t, exists, "Rule should not be present in asyncRuleCache")
 			assert.NoErrorf(t, err, "getRuleFromAsyncCache should not return any error")
-
-			elapsedTime := time.Since(start)
-			assert.GreaterOrEqualf(t, int64(elapsedTime)/int64(time.Millisecond), int64(5), "rule should be there for at least 5ms")
+			// Delta accounts for both deletion time of rule after adding it to deleteQueue and time taken to run the condition function once the poll happens.
+			assert.InDeltaf(t, int64(elapsedTime)/int64(time.Millisecond), int64(a.deleteInterval)/int64(time.Millisecond), 100, "rule should be in cache for about %v", a.deleteInterval)
 		})
 	}
 }

--- a/pkg/agent/controller/networkpolicy/networkpolicy_controller_test.go
+++ b/pkg/agent/controller/networkpolicy/networkpolicy_controller_test.go
@@ -52,7 +52,7 @@ func (g *antreaClientGetter) GetAntreaClient() (versioned.Interface, error) {
 func newTestController() (*Controller, *fake.Clientset, *mockReconciler) {
 	clientset := &fake.Clientset{}
 	ch := make(chan v1beta2.PodReference, 100)
-	controller, _ := NewNetworkPolicyController(&antreaClientGetter{clientset}, nil, nil, "node1", ch, true, testDeleteInterval)
+	controller, _ := NewNetworkPolicyController(&antreaClientGetter{clientset}, nil, nil, "node1", ch, true, testAsyncDeleteInterval)
 	reconciler := newMockReconciler()
 	controller.reconciler = reconciler
 	return controller, clientset, reconciler

--- a/pkg/agent/controller/networkpolicy/reconciler_test.go
+++ b/pkg/agent/controller/networkpolicy/reconciler_test.go
@@ -173,7 +173,7 @@ func TestReconcilerForget(t *testing.T) {
 					mockOFClient.EXPECT().UninstallPolicyRuleFlows(ofID)
 				}
 			}
-			r := newReconciler(mockOFClient, ifaceStore, testDeleteInterval)
+			r := newReconciler(mockOFClient, ifaceStore, testAsyncDeleteInterval)
 			for key, value := range tt.lastRealizeds {
 				r.lastRealizeds.Store(key, value)
 			}
@@ -517,7 +517,7 @@ func TestReconcilerReconcile(t *testing.T) {
 			for i := 0; i < len(tt.expectedOFRules); i++ {
 				mockOFClient.EXPECT().InstallPolicyRuleFlows(gomock.Any())
 			}
-			r := newReconciler(mockOFClient, ifaceStore, testDeleteInterval)
+			r := newReconciler(mockOFClient, ifaceStore, testAsyncDeleteInterval)
 			if err := r.Reconcile(tt.args); (err != nil) != tt.wantErr {
 				t.Fatalf("Reconcile() error = %v, wantErr %v", err, tt.wantErr)
 			}
@@ -627,7 +627,7 @@ func TestReconcilerBatchReconcile(t *testing.T) {
 			mockOFClient := openflowtest.NewMockClient(controller)
 			mockOFClient.EXPECT().IsIPv4Enabled().Return(true).AnyTimes()
 			mockOFClient.EXPECT().IsIPv6Enabled().Return(false).AnyTimes()
-			r := newReconciler(mockOFClient, ifaceStore, testDeleteInterval)
+			r := newReconciler(mockOFClient, ifaceStore, testAsyncDeleteInterval)
 			if tt.numInstalledRules > 0 {
 				// BatchInstall should skip rules already installed
 				r.lastRealizeds.Store(tt.args[0].ID, newLastRealized(tt.args[0]))
@@ -843,7 +843,7 @@ func TestReconcilerUpdate(t *testing.T) {
 			if len(tt.expectedDeletedTo) > 0 {
 				mockOFClient.EXPECT().DeletePolicyRuleAddress(gomock.Any(), types.DstAddress, gomock.Eq(tt.expectedDeletedTo), priority)
 			}
-			r := newReconciler(mockOFClient, ifaceStore, testDeleteInterval)
+			r := newReconciler(mockOFClient, ifaceStore, testAsyncDeleteInterval)
 			if err := r.Reconcile(tt.originalRule); (err != nil) != tt.wantErr {
 				t.Fatalf("Reconcile() error = %v, wantErr %v", err, tt.wantErr)
 			}
@@ -1349,7 +1349,7 @@ func TestReconcilerReconcileIPv6Only(t *testing.T) {
 			for i := 0; i < len(tt.expectedOFRules); i++ {
 				mockOFClient.EXPECT().InstallPolicyRuleFlows(gomock.Any())
 			}
-			r := newReconciler(mockOFClient, ifaceStore, testDeleteInterval)
+			r := newReconciler(mockOFClient, ifaceStore, testAsyncDeleteInterval)
 			if err := r.Reconcile(tt.args); (err != nil) != tt.wantErr {
 				t.Fatalf("Reconcile() error = %v, wantErr %v", err, tt.wantErr)
 			}
@@ -1752,7 +1752,7 @@ func TestReconcilerReconcileDualStack(t *testing.T) {
 			for i := 0; i < len(tt.expectedOFRules); i++ {
 				mockOFClient.EXPECT().InstallPolicyRuleFlows(gomock.Any())
 			}
-			r := newReconciler(mockOFClient, ifaceStore, testDeleteInterval)
+			r := newReconciler(mockOFClient, ifaceStore, testAsyncDeleteInterval)
 			if err := r.Reconcile(tt.args); (err != nil) != tt.wantErr {
 				t.Fatalf("Reconcile() error = %v, wantErr %v", err, tt.wantErr)
 			}

--- a/test/e2e/flowexporter_test.go
+++ b/test/e2e/flowexporter_test.go
@@ -56,7 +56,7 @@ func TestFlowExporter(t *testing.T) {
 	}
 
 	// Add NetworkPolicy between two iperf Pods.
-	np1, err := data.createNetworkPolicy("test-networkpolicy-ingress", &networkingv1.NetworkPolicySpec{
+	np1, err := data.createNetworkPolicy("test-flow-exporter-networkpolicy-ingress", &networkingv1.NetworkPolicySpec{
 		PodSelector: metav1.LabelSelector{},
 		PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeIngress},
 		Ingress: []networkingv1.NetworkPolicyIngressRule{{
@@ -77,7 +77,7 @@ func TestFlowExporter(t *testing.T) {
 			t.Fatalf("Error when deleting network policy: %v", err)
 		}
 	}()
-	np2, err := data.createNetworkPolicy("test-networkpolicy-egress", &networkingv1.NetworkPolicySpec{
+	np2, err := data.createNetworkPolicy("test-flow-exporter-networkpolicy-egress", &networkingv1.NetworkPolicySpec{
 		PodSelector: metav1.LabelSelector{},
 		PolicyTypes: []networkingv1.PolicyType{networkingv1.PolicyTypeEgress},
 		Egress: []networkingv1.NetworkPolicyEgressRule{{
@@ -196,10 +196,10 @@ func checkRecordsWithPodIPs(t *testing.T, data *TestData, podAIP string, podBIP 
 				// One of them has no bytes and we ignore that flow record.
 				if !strings.Contains(record, "octetDeltaCount: 0") {
 					// Check if records have both ingress and egress network policies.
-					if !strings.Contains(record, "test-networkpolicy-ingress") {
+					if !strings.Contains(record, "test-flow-exporter-networkpolicy-ingress") {
 						t.Fatalf("Records does not have NetworkPolicy name with ingress rule")
 					}
-					if !strings.Contains(record, "test-networkpolicy-egress") {
+					if !strings.Contains(record, "test-flow-exporter-networkpolicy-egress") {
 						t.Fatalf("Records does not have NetworkPolicy name with egress rule")
 					}
 				}

